### ISSUE TITLE
quality of the output *.wav file

### ISSUE THREAD

**Issue by HaoyuanLuo** [keithito/tacotron#86]
Thanks for your@keithito awesome job of Tacotron. 
With many tries of tacotron in English and Mandarin. I got some acceptable results even in personlize synthesis.

After listening the samples from Google and our issues. I found that the high quality results always are in fixed duration. So I wonder if there is a way to imporve the quality of the output wav file with fixeding the output duration? 

I've read a paper named **Emotional End-to-End Neural Speech synthesizer** which has mentioned some way to improve the quality of the output audio based on this implement. So I am tring 
 MonotonicAttension and adding residual connections. I'm not sure whether I can get the same results as my expectation.

**Comment by keithito** [comment]
The fixed output length is because the model doesn't learn how to stop emitting output. This implementation also generates fixed output length, but there's a preprocessing step to trim silence off the end.

Yes, I think using monotonic attention may be helpful. Also, replacing Griffin-Lim with something like Wavenet as suggested [here](https://github.com/keithito/tacotron/issues/83) would also greatly improve quality.

**Comment by yodakohl** [comment]
I'm currently at step 60k using monotonic attention and scheduled sampling (LJSpeech). 

Using the upstream implementation i haven't been able to reproduce the results from keithito. There was always a big difference between test and training quality. Now I get a very clear attention and the pronounciation in the evaluation is good, however the audio quality is still very rough.


![step-60000-align](https://user-images.githubusercontent.com/819019/34118477-359478bc-e41f-11e7-9e14-aaa9493b9dc0.png)

[Eval Audio Step 60k](https://soundcloud.com/user-837898227/eval-60000-1)
[Training Audio Step 60k](https://soundcloud.com/user-837898227/step-60000-audio)

**Comment by wotulong** [comment]
I think the attention is not clear, because a clear attention means the color of most pixel in the picture is yellow, which means the value is to 1. And your plot shows that most value is smaller than 0.2.

**Comment by nukes** [comment]
![attention40k](https://user-images.githubusercontent.com/4599114/34150308-f56e33b2-e4e2-11e7-8956-171160bdf6eb.png)

![attention22k](https://user-images.githubusercontent.com/4599114/34150304-f317ad28-e4e2-11e7-97ae-543473f1b6f4.png)
i think 60k steps is still not sufficient. i also tried mono-attention on nancy corpus and got good results.
I think you should be careful about your monotonic attention parameter settings. Very important!
Also, the monotonic attention can help us stop decoding once we know we know that we have just attended the last character. so we do not need to reconstruct silence wave.
the alignment below were got at 4k, 22k, 40k, 87k ,100k steps respectively.

![att4k](https://user-images.githubusercontent.com/4599114/34150516-80477c3c-e4e3-11e7-8703-021bed41fcd6.png)
![attention22k](https://user-images.githubusercontent.com/4599114/34150518-80ee0a02-e4e3-11e7-8b3e-ff0103ab2b37.png)
![attention40k](https://user-images.githubusercontent.com/4599114/34150521-82bf4e18-e4e3-11e7-8b42-7f9ada1e6de9.png)
![att87k](https://user-images.githubusercontent.com/4599114/34150524-83b93324-e4e3-11e7-9976-670015aefa26.png)
![att100k](https://user-images.githubusercontent.com/4599114/34150531-879c2a3c-e4e3-11e7-883b-eb61112fc335.png)

**Comment by nukes** [comment]
@HaoyuanLuo did you try feed the mandarin model with character or pinyin separated by shenmu and yunmu ? what about the convergence under these two situation ?

**Comment by HaoyuanLuo** [comment]
@nukes Your attention alignment is the best one I've ever seen in a so fast convergence. I will make it as 
 my  goal.
For Mandarin ,I've tried many kind of ways.I didn't get results with shenmu yunmu and I'm planing to try it again. Character way converges at 15K step. I think shenmuyunmu will be better than character.

**Comment by nukes** [comment]
@HaoyuanLuo  I am working on characters for mandarin, and keep in touch!

**Comment by HaoyuanLuo** [comment]
@nukes That's great!

**Comment by nukes** [comment]
@HaoyuanLuo  Hi, what kind of corpus you are using for mandarin synthesis? i can not find a good corpus on the web. Any suggestions ?

**Comment by HaoyuanLuo** [comment]
@nukes I'm sorry for the late reply. We got Chinese Data from the Internet FM like XImalaya or Lizhi FM. We use Baidu's or Google's ASR to  mark audio to text. If we want a great corpus, we should proofread the data one by one. But I think the  automatic generation result is enough to make a test. 

Besides, I have one more question about the parameters of AttentionNet. I tried some different scale of parameters last week after getting your advice, the speed of convergence can arrive at the same stage of yours. However, the attention value is always near 0.5 after 100K steps, much worse than your 0.8. Would you like to share some way to make the value promote? I'll be very hungered for learning that.

**Comment by nukes** [comment]
@HaoyuanLuo  Thanks for your suggestions. for the attention part, a base trick to make the network learn faster is that you should make sure that the output distribution for each dim is zero mean and std diviation.
i did not follow the original code from this repo for normalization.  All in all, normalization is important!

**Comment by HaoyuanLuo** [comment]
@nukes Thanks ever so much for your advice, I really appreciate it. I will learn the relative knowledge as you said and do a couple of  experiments.

**Comment by t3t3t3** [comment]
Hi all, 

I find that quality of audio produced by synthesizer.py is lower than the audio dump out by train.py during training (especially for long sentence).

On closer look, I find some differences between synthesizer.py & train.py:
1. synthesizer generate audio of length defined by max_iters in hparams (and then cut of the silence tail), while train.py is not (the audio length is known from training data)
2. synthesizer user audio.inv_spectrogram_tensorflow ; while train.py use audio.inv_spectrogram
3. synthesizer use input sequence from text without padding of _  , while train.py add padding of _ ti input sequence (? !)

I have not look into why train.py need padding of _ , but is it the main cause for the superior audio quality generated by train.py?

**Comment by zuoxiang95** [comment]
In training process, it uses teacher forcing so the RNN gets fed the previous ground truth output as input when predicting the current output.  When you generate audio in eval process,  RNN get the previous output as input. So, the generate audio in eval process would be worse than train process.

**Comment by TranTheTrung** [comment]
Thank you for your explanation!

**Comment by StanDex** [comment]
Guys, please help! I can't launch any checkpoint - a problem with path find... https://github.com/keithito/tacotron/issues/94

**Comment by begeekmyfriend** [comment]
Hi guys, I wonder what kind of symbolic characters do you design for text encoding? Just like [alphabet plus number](https://github.com/begeekmyfriend/tacotron/blob/master/text/symbols.py#L12) or [pinyin phonemes](https://github.com/begeekmyfriend/tacotron/blob/5d881fab3823d875ac10a8fdd796048876b17ebc/text/symbols.py#L9) ? @HaoyuanLuo @nukes
